src/main/webapp/teacher/controlGameInstance/controlGameInstance.js

```diff
@@ -188,22 +203,131 @@ async function updateCurrentQuestion({ gameId, currentQuestionId, isCurrentQuest
   activeQuestionNumberElement.innerText = "Students are seeing question with ID: " + (currentQuestionId);
 }
 
+// Queries and returns the currentQuestion object
+function queryCurrentQuestion({ gameId, currentQuestionId }) {
+  return db.collection('games').doc(gameId).collection('questions').doc(currentQuestionId).get().then(function(doc) {
+    if (doc.exists) {
+      return doc.data()
+    }
+  });
+}
+
+// Init the listener to a change in the current question's answers
+function initQuestionAnswerStatsListener({ gameInstanceId, currentQuestionId } = {}) {
```

Comment:
What does `{ gameInstanceId, currentQuestionId } = {}` mean?

Reply:
This is a common way to have named parameters with defaults in JS. Calling this function with an argument `{ gameInstanceId: 123, currentQuestionId: 5 }` will result in the argument object getting destructured and its values will be assigned to the identifiers `gameInstanceId` and `currentQuestionId` accordingly.